packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -463,13 +463,31 @@ export class Database {
     }
   }
 
-  public indexContentByPaths = async (
-    documentPaths: string[],
-    collection:
-      | CollectionFieldsWithNamespace<true>
-      | CollectionTemplatesWithNamespace<true>
-  ) => {
-    await _indexContent(this, documentPaths, collection)
+  public indexContentByPaths = async (documentPaths: string[]) => {
+    const pathsByCollection: Record<string,string[]> = {}
+    const nonCollectionPaths: string[] = []
+    const collections: Record<string,CollectionFieldsWithNamespace<true> | CollectionTemplatesWithNamespace<true>> = {}
+    const tinaSchema = await this.getSchema()
+    for (const documentPath of documentPaths) {
+      const collection = tinaSchema.schema.collections.find(
+        (collection) => documentPath.startsWith(collection.path)
```

Comment:
This is probably a change for the better, if/when we want to actually support the `matches` property in our collection config it'll be easier to do so here.